risc0/zkp/src/core/blake2b.rs

```diff
@@ -0,0 +1,139 @@
+// Copyright 2023 RISC Zero, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! A Blake2b HashSuite.
+use alloc::boxed::Box;
+use alloc::vec::Vec;
+use core::marker::PhantomData;
+
+use blake2::{digest::{Update, VariableOutput}, VarBlake2b};
+
+use risc0_core::field::baby_bear::{BabyBear, BabyBearElem, BabyBearExtElem};
+use risc0_core::field::ExtElem;
+
+use crate::core::config::{ConfigHash, ConfigRng};
+
+use super::config::HashSuite;
+use super::digest::Digest;
+
+/// Hash function trait.
+pub trait Blake2bHasher {
+    /// A function producing a hash from a list of u8.
+    fn blake2b<T: AsRef<[u8]>>(data: T) -> [u8; 32];
+}
+
+/// Implementation of blake2b using CPU.
```

Comment:
The reason we use a trait for the underlying hasher in the Sha example is to allow the guest to use a special Sha256 acceleration circuit.  You are welcome to similarly abstract Blake2, but you can also probably skip it if that's simpler, since it will probably be a while before we add a Blake2 accelerator.

Reply:
Isn't it already abstracted in a similar way? We are using it to replace the blake2b implementation with a Casper host one.
I've changed the trait's name to `Blake2b` to reflect this better.